pkg/publish/daemon.go

```diff
@@ -47,9 +48,43 @@ func (d *demon) Publish(br build.Result, s string) (name.Reference, error) {
 	// https://github.com/google/go-containerregistry/issues/212
 	s = strings.ToLower(s)
 
-	// There's no way to write an index to a daemon, so attempt to downcast it to an image.
-	img, ok := br.(v1.Image)
-	if !ok {
+	// There's no way to write an index to a kind, so attempt to downcast it to an image.
+	var img v1.Image
+	switch i := br.(type) {
+	case v1.Image:
+		img = i
+	case v1.ImageIndex:
+		im, err := i.IndexManifest()
```

Comment:
Does this seem like a helper we should add to ggcr? `idx.ImageByArch(os,arch)`?

Reply:
Yeah probably.  I copied this from kind.

I think I left a similar comment on @jonjohnsonjr's original PR.

Reply:
fyi https://github.com/google/go-containerregistry/pull/797